code/modules/mob/living/simple_animal/friendly/corgi.dm

```diff
@@ -493,50 +500,112 @@
 		return 0 //under effects of time magick
 
 	..()
-
-	//Feeding, chasing food, FOOOOODDDD
-	if(!stat && !resting && !locked_to && (ckey == null))
-		turns_since_scan++
-		if(turns_since_scan > 5)
-			turns_since_scan = 0
-			if((movement_target) && !(isturf(movement_target.loc) || ishuman(movement_target.loc) ))
-				movement_target = null
-				stop_automated_movement = 0
-			if( !movement_target || !(movement_target.loc in oview(src, 3)) )
-				movement_target = null
-				stop_automated_movement = 0
-				for(var/obj/item/weapon/reagent_containers/food/snacks/S in oview(src,3))
-					if(isturf(S.loc) || ishuman(S.loc))
-						movement_target = S
+	
+	if(!stat && !resting && !locked_to && (ckey == null)) //Behavior mechanisms (om nom :3)
+		if(ian_status == IDLE)
+			get_target()
+			stop_automated_movement = 0		
+			
+		else if(ian_status == BEGIN_FOOD_HUNTING)
+			ian_status = FOOD_HUNTING
+			spawn(0) // Separate process
+				stop_automated_movement = 1
+				var/failedsteps
+				while(1)
```

Comment:
highly fucking illegal

Reply:
Eh there are very clear break conditions. Maybe add an explicit timeout I don't know

Reply:
Are you referring to the while(true) loops? Like shifty said the break conditions will always trigger, but I can improve it in case of a fringe case.

Reply:
just do `while(failedsteps <= 3)`